### ISSUE TITLE
When using library only with udp, still need to provide address in http format

### ISSUE THREAD

**Issue by abadyan-vonage** [influxdata/influxdb-java#279]
If I don't provide a URL with an "http://" prefix, I get a NullReferenceException.
This is not relevant for udp.

**Comment by majst01** [comment]
You are right, do you have an idea howto instantiate the InfluxDBImpl with the Factory to be safe for both cases ? If yes, please write a PR.

**Comment by asashour** [comment]
If InfluxDbImpl is initiated with only UDP hostname, then we can't use `.query()`, `.ping()' or any HTTP functions. The current design mixes both HTTP and UDP hostname in the URL.

I would suggest separating the UDP from HTTP, and we have HttpInfluxDB and UdpInfuxDB, each one with different methods, e.g. `.query()` in HTTP and `.write(udp)` in UDP implementation. Both extends InfluxDB interface, which has methods for writing values.

Obviously, this is a breaking change.
